src/main/java/at/ac/tuwien/kr/alpha/common/terms/FunctionTerm.java

```diff
@@ -116,23 +115,22 @@ public int hashCode() {
 
 	@Override
 	public int compareTo(Term o) {
+		if (o instanceof ConstantTerm) {
+			return 1;
+		}
 		if (!(o instanceof FunctionTerm)) {
-			throw new ClassCastException();
+			throw new UnsupportedOperationException("Can only compare function term to function term or constant term.");
```

Comment:
According to the ASP Core2 standard, all kinds of terms can be compared. Our previous orders did not take this into account, but we should use/support the official term ordering soonish.

Reply:
Yeah, from what I see, the changes in this branch improve the situation, but we're not 100% there yet.